File: analytics-service/src/main/java/com/emergency/analytics/model/AnalyticsEvent.java
package com.emergency.analytics.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "analytics_events")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AnalyticsEvent {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "event_type", nullable = false)
    private String eventType;

    @Column(name = "incident_id")
    private Long incidentId;

    @Column(name = "incident_type")
    private String incidentType;

    @Column
    private String region;

    @Column(name = "response_time_minutes")
    private Double responseTimeMinutes;

    @Column(name = "responder_type")
    private String responderType;

    @Column(name = "responder_id")
    private Long responderId;

    @Column(columnDefinition = "TEXT")
    private String details;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime timestamp;
}

File: nodejs-migration/analytics-service/src/entities/AnalyticsEvent.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
} from 'typeorm';

export enum IncidentType {
  FIRE = 'FIRE',
  MEDICAL = 'MEDICAL',
  ACCIDENT = 'ACCIDENT',
  HAZMAT = 'HAZMAT',
  OTHER = 'OTHER',
}

@Entity('analytics_events')
export class AnalyticsEvent {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column()
  eventType!: string;

  @Column({ type: 'enum', enum: IncidentType, nullable: true })
  incidentType?: IncidentType;

  @Column({ nullable: true })
  region?: string;

  @Column({ nullable: true })
  responseTimeMinutes?: number;

  @Column({ nullable: true })
  incidentId?: number;

  @Column({ type: 'simple-json', nullable: true })
  metadata?: any;

  @CreateDateColumn()
  timestamp!: Date;
}
